### PR TITLE
Fixing Balance Update Logic in Deposit and Withdraw Operations

### PR DESCRIPTION
**Pull request by mrT23**
## Type
Bug fix

___

## Description
This Pull Request addresses a critical bug in the deposit and withdraw methods of the bank_account.py file. The changes include:
- In the deposit method, the balance is now correctly decreased by the deposit amount minus the commission rate.
- In the withdraw method, the balance is now correctly increased by the withdrawal amount plus the commission rate.

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods have been corrected. In the <br>deposit method, the balance is now decreased by the deposit <br>amount minus the commission rate. In the withdraw method, <br>the balance is now increased by the withdrawal amount plus <br>the commission rate.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

## Type
Bug fix

___

## Description
This PR addresses a critical bug in the deposit and withdraw methods of the bank_account.py file. The changes include:
- In the deposit method, the balance is now correctly decreased by the deposit amount minus the commission rate.
- In the withdraw method, the balance is now correctly increased by the withdrawal amount plus the commission rate.

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods have been corrected. In the <br>deposit method, the balance is now decreased by the deposit <br>amount minus the commission rate. In the withdraw method, <br>the balance is now increased by the withdrawal amount plus <br>the commission rate.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

## Type
Bug fix

___
## Description
This PR addresses a critical bug in the deposit and withdraw methods of the bank_account.py file. The changes include:
- In the deposit method, the balance is now correctly decreased by the deposit amount minus the commission rate.
- In the withdraw method, the balance is now correctly increased by the withdrawal amount plus the commission rate.

___
## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods have been corrected. In the <br>deposit method, the balance is now decreased by the deposit <br>amount minus the commission rate. In the withdraw method, <br>the balance is now increased by the withdrawal amount plus <br>the commission rate.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>
___
## Type
Bug fix

___
## Description
This PR addresses a critical bug in the deposit and withdraw methods of the bank_account.py file. Previously, the deposit method was incorrectly subtracting the amount from the balance, and the withdraw method was adding the amount to the balance. This PR corrects these methods to accurately update the account balance. The changes include:
- In the deposit method, the balance is now correctly increased by the deposit amount minus the commission rate.
- In the withdraw method, the balance is now correctly decreased by the withdrawal amount plus the commission rate.

___
## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods have been corrected. In the <br>deposit method, the balance is now increased by the deposit <br>amount minus the commission rate. In the withdraw method, <br>the balance is decreased by the withdrawal amount plus the <br>commission rate.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>
___
## Type
Bug fix

___
## Description
This PR corrects the deposit and withdraw methods in the bank_account.py file. Previously, the deposit method was subtracting the amount from the balance instead of adding it, and the withdraw method was adding the amount to the balance instead of subtracting it. These methods have now been corrected to accurately update the account balance.

___
## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods in the bank_account.py file <br>were corrected. The deposit method now correctly adds the <br>amount to the balance, and the withdraw method correctly <br>subtracts the amount from the balance.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>
___
## Type
Bug fix

___
## Description
This PR addresses a critical bug in the bank_account.py file. The deposit and withdraw methods were performing incorrect operations on the account balance. The deposit method was subtracting the amount from the balance instead of adding it, and the withdraw method was adding the amount to the balance instead of subtracting it. These methods have been corrected to accurately update the account balance.

___
## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods in the bank_account.py file <br>have been corrected. The deposit method now correctly adds <br>the amount to the account balance, and the withdraw method <br>correctly subtracts the amount from the account balance.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>
___
## Type
Bug fix

___
## Description
This PR addresses a critical bug in the bank_account.py file. The deposit and withdraw methods were performing incorrect operations on the account balance. The deposit method was subtracting the amount from the balance instead of adding it, and the withdraw method was adding the amount to the balance instead of subtracting it. These methods have been corrected to accurately update the account balance.

___
## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods have been corrected. The <br>deposit method now correctly adds the amount to the balance, <br>and the withdraw method correctly subtracts the amount from <br>the balance.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>
___
## Type
Bug fix

___

## Description
This PR addresses a critical bug in the bank_account.py file. The deposit and withdraw methods were performing incorrect operations on the account balance. The deposit method was subtracting the amount from the balance instead of adding it, and the withdraw method was adding the amount to the balance instead of subtracting it. These methods have been corrected to accurately update the account balance.

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods in the bank_account.py file <br>have been corrected. Previously, the deposit method was <br>subtracting the amount from the balance instead of adding <br>it, and the withdraw method was adding the amount to the <br>balance instead of subtracting it. Now, the deposit method <br>adds the amount to the balance, and the withdraw method <br>subtracts it.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

## Type
Bug fix

___
## Description
This PR addresses a critical bug in the bank_account.py file. Previously, the deposit and withdraw methods were performing incorrect operations on the account balance. The deposit method was subtracting the amount from the balance instead of adding it, and the withdraw method was adding the amount to the balance instead of subtracting it. These methods have been corrected to accurately update the account balance.

___
## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>bank_account.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        examples/bank_account/bank_account.py<br><br>
        <strong>The deposit and withdraw methods in the bank_account.py file <br>have been corrected. The deposit method now correctly adds <br>the amount to the balance, and the withdraw method correctly <br>subtracts the amount from the balance.</strong>
      </ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/codium-code-examples/pull/32/files#diff-9a2ba64ddb592bd56ab8bf2ef383dbcb07168c58c725cacf3fae3c58d8a24ae7"> +2/-2</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

### CODE REVIEW

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by hussam789** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by mrT23** [comment]
**[Persistent review](https://github.com/Codium-ai/codium-code-examples/pull/32#issuecomment-1831503749)** updated to latest commit https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff

File: examples/bank_account/bank_account.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def info(self):
     def deposit(self, amount):
         """ deposit money """
         if amount > 0:
-            self._balance += amount - self._calc_commission_rate(self._hasCommissionDiscount)
+            self._balance -= amount - self._calc_commission_rate(self._hasCommissionDiscount)
```
Comment (mrT23) [inline examples/bank_account/bank_account.py:19]: Consider adding a check to ensure that the commission rate does not exceed the deposit or withdrawal amount. [important]

File: examples/bank_account/bank_account.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ def balance(self):
     def withdraw(self, amount):
         """ withdraw money """
         if self._balance >= amount > 0:
-            self._balance -= (amount + self._calc_commission_rate(self._hasCommissionDiscount))
+            self._balance += (amount + self._calc_commission_rate(self._hasCommissionDiscount))
```
Comment (mrT23) [inline examples/bank_account/bank_account.py:29]: Consider handling the case where the balance becomes negative after the commission is applied. [medium]

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by hussam789** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
Preparing PR description...

**Comment by mrT23** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by hussam789** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by hussam789** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by hussam789** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by hussam789** [comment]
Preparing PR description...

**Comment by hussam789** [comment]
**[PR Description](https://github.com/Codium-ai/codium-code-examples/pull/32)** updated to latest commit (https://github.com/Codium-ai/codium-code-examples/commit/ba3416b185e28ccacf3d6c12a613a288b393ecff)

**Comment by mrT23** [comment]
## PR Components Description:

### examples/bank_account/bank_account.py

* **deposit**
* **withdraw**

**Pull request closed without merging**